OpenDreamRuntime/DreamMapManager.cs

```diff
@@ -139,21 +139,25 @@ public struct Cell {
                 DreamPath areaType = cellDefinition.Area != null ? _dreamManager.ObjectTree.Types[cellDefinition.Area.Type].Path : _defaultArea;
                 DreamObject area = GetArea(areaType);
 
+                int x = block.X + blockX - 1;
+                int y = block.Y + block.Height - blockY;
+
                 DreamObject turf;
                 if (cellDefinition.Turf != null) {
                     turf = CreateMapObject(cellDefinition.Turf);
+                    SetTurf(x, y, block.Z, turf);
+                    SetArea(x, y, block.Z, area);
+                    turf.InitSpawn(new DreamProcArguments(new() {DreamValue.Null}));
                 } else {
                     turf = _dreamManager.ObjectTree.CreateObject(_defaultTurf);
+                    SetTurf(x, y, block.Z, turf);
+                    SetArea(x, y, block.Z, area);
                     turf.InitSpawn(new DreamProcArguments(null));
                 }
```

Comment:
This repeated code could be moved outside the if/else
```suggestion
                } else {
                    turf = _dreamManager.ObjectTree.CreateObject(_defaultTurf);
                }
                
                SetTurf(x, y, block.Z, turf);
                SetArea(x, y, block.Z, area);
                turf.InitSpawn(new DreamProcArguments(null));
                    
```